torch_xla/csrc/tensor.cpp

```diff
@@ -249,17 +249,18 @@ void XLATensor::SetShardingSpec(const ShardingSpec& sharding, bool overwrite) {
   // Existing annotation must be cleared explicitly. We do not clear and
   // overwrite the existing sharding on the user's behalf. This is a no-op if
   // the same sharding already applied.
-  if (!sharding_spec() || overwrite ||
-      sharding_spec()->sharding.type() == xla::OpSharding::REPLICATED ||
-      sharding_spec()->sharding.type() == xla::OpSharding::UNKNOWN) {
+  ShardingSpecPtr current_sharding = sharding_spec();
+  if (!current_sharding || overwrite ||
+      current_sharding->sharding.type() == xla::OpSharding::REPLICATED ||
+      current_sharding->sharding.type() == xla::OpSharding::UNKNOWN) {
```

Comment:
this is just a clean up right? I don't see any functionality change but want to double check.

Reply:
It's a clean-up